src/axolotl/integrations/nvfp4/kernels/nvfp4_fused_ce.py

```
@@ -0,0 +1,301 @@
+
+class _FusedFP4CrossEntropy(torch.autograd.Function):
+    """Tiled lm_head(FP4) -> fp32 logsumexp/gather -> CE, no ``[M, V]`` logits.
+
+    backward recomputes the softmax tile-by-tile from the saved logsumexp.
+    ``grad_scale`` (1/num_items for grad-accum, else 1/valid_count) is pre-folded so
+    backward is a pure function of the saved tensors. lm_head is frozen, no wgrad.
+    """
+
+    @staticmethod
+    def forward(ctx, hidden, store, labels, ignore_index, scale, grad_scale):
+        M, H = hidden.shape
```

Ruff (0.15.20)

[warning] Unpacked variable `H` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)